examples/cpp-simple/server.cc

```diff
@@ -114,6 +114,8 @@ int main() {
       std::cout << "...marked for Shutdown\n" << std::flush;
     }
   }
+  health.detach();
```

Comment:
It's preferable to join rather than detach, when possible. http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#cp26-dont-detach-a-thread

Can the thread bodies detect that the SDK was shutdown and then exit gracefully? That way main can join the threads here instead of detach them.

Reply:
Yes, I will include that shutdown trigger for threads into example code.